src/Microsoft.DncEng.SecretManager/StorageTypes/AzureKeyVault.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Security.KeyVault.Secrets;
+using Microsoft.DncEng.CommandLineLib;
+using Microsoft.DncEng.CommandLineLib.Authentication;
+
+namespace Microsoft.DncEng.SecretManager.StorageTypes
+{
+    [Name("azure-key-vault")]
+    public class AzureKeyVault : StorageLocationType
+    {
+        private static readonly string _nextRotationOnTag = "next-rotation-on";
+        private readonly TokenCredentialProvider _tokenCredentialProvider;
+        private readonly Guid _subscription;
+        private readonly string _vaultName;
+
+        public AzureKeyVault(IReadOnlyDictionary<string, string> parameters, TokenCredentialProvider tokenCredentialProvider) : base(parameters)
+        {
+            _tokenCredentialProvider = tokenCredentialProvider;
+            ReadRequiredParameter("subscription", ref _subscription);
+            ReadRequiredParameter("name", ref _vaultName);
+        }
+
+        private async Task<SecretClient> CreateSecretClient()
+        {
+            var creds = await _tokenCredentialProvider.GetCredentialAsync();
+            return new SecretClient(new Uri($"https://{_vaultName}.vault.azure.net/"), creds);
```

Comment:
Didn't want to do a factory pattern here?

Reply:
This type is a thin wrapper around the SecretClient, I didn't think an even smaller wrapper would be useful at all. This type doesn't have any real logic in it, just data mapping, so it can get full coverage from just integration tests.